components/script/dom/cssstyledeclaration.rs

```diff
@@ -400,7 +424,8 @@ impl CSSStyleDeclarationMethods for CSSStyleDeclaration {
 
     // https://dev.w3.org/csswg/cssom/#the-cssstyledeclaration-interface
     fn IndexedGetter(&self, index: u32) -> Option<DOMString> {
-        self.owner.with_block(|pdb| {
+        self.owner.with_mut_block(|pdb| {
+            pdb.deduplicate();
```

Comment:
So I'm somewhat concerned about this, since this means that all these calls will be not only `O(n)`, but also require a reallocation unconditionally.

I haven't been following all this business closely, but I think deduplicating only if we've seen duplicated properties during parsing should be good enough?